client/src/components/MoveIn.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import MoveInContent from './MoveInContent.jsx';
+ 
+class MoveIn extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      show: false,
+      date: props.date || null
+    };
+    this.set = props.set;
+  }
+
+  handleShowChange() {
+    const show = !this.state.show;
+    this.setState({ show });
+  }
+
+  setDate(date) {
+    if (date) {
+      this.setState({ date, show: false });
+      this.set(date);
```

Comment:
What is the purpose of keeping track of date in two places? Why not use one source of truth from the parent? 

Additionally, because `date` is an object, does modifying one instance of it alter the other? In other words, is the date value copied on line 9, or are both variables pointing to the same value in memory?

Reply:
Hmm, good point, I should rethink my approach here...